pkg/server/plugin/upstreamca/disk/disk.go

```diff
@@ -108,57 +113,84 @@ func (p *DiskPlugin) SubmitCSR(ctx context.Context, request *upstreamca.SubmitCS
 
 	return &upstreamca.SubmitCSRResponse{
 		SignedCertificate: &upstreamca.SignedCertificate{
-			CertChain: cert.Raw,
-			Bundle:    upstreamCert.Raw,
+			CertChain: append(cert.Raw, upstreamCerts.certChain...),
+			Bundle:    upstreamCerts.trustBundle,
 		},
 	}, nil
 }
 
-func (p *DiskPlugin) reloadCA() (*x509svid.UpstreamCA, *x509.Certificate, error) {
+func (p *DiskPlugin) reloadCA() (*x509svid.UpstreamCA, *caCerts, error) {
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 
-	upstreamCA, upstreamCert, err := p.loadUpstreamCAAndCert(p.config)
+	upstreamCA, upstreamCerts, err := p.loadUpstreamCAAndCerts(p.config)
 	switch {
 	case err == nil:
 		p.upstreamCA = upstreamCA
-		p.cert = upstreamCert
+		p.certs = upstreamCerts
 	case p.upstreamCA != nil:
 		upstreamCA = p.upstreamCA
-		upstreamCert = p.cert
+		upstreamCerts = p.certs
 	default:
 		return nil, nil, fmt.Errorf("no cached CA and failed to load CA: %v", err)
 	}
 
-	return upstreamCA, upstreamCert, nil
+	return upstreamCA, upstreamCerts, nil
 }
 
-func (p *DiskPlugin) loadUpstreamCAAndCert(config *Configuration) (*x509svid.UpstreamCA, *x509.Certificate, error) {
+func (p *DiskPlugin) loadUpstreamCAAndCerts(config *Configuration) (*x509svid.UpstreamCA, *caCerts, error) {
 	key, err := pemutil.LoadPrivateKey(config.KeyFilePath)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	cert, err := pemutil.LoadCertificate(config.CertFilePath)
+	certs, err := pemutil.LoadCertificates(config.CertFilePath)
 	if err != nil {
 		return nil, nil, err
 	}
+	// pemutil guarantees at least 1 cert
+	caCert := certs[0]
+
+	var trustBundle []byte
+	if config.BundleFilePath == "" {
+		// If there is no bundle path configured then we assume the chain
+		// including the root comes from cert_file_path. The final cert
+		// is the root that should be trusted. Additionally, it can be dropped
+		// from the cert chain to avoid wastefully distributing it in the chain
+		// of an SVID as well as the trust bundle.
+		trustBundle = certs[len(certs)-1].Raw
+		certs = certs[:len(certs)-1]
```

Comment:
i think we want to be careful about using a cert chain here to represent the upstream ca, and may want to put as much as possible into the trust bundle

from my reading, if you're joining an upstream ca pki (which i believe we are), the rest of the cert chain is tossed out

https://github.com/spiffe/spire/blob/5ef1c258bb27e0abf1ef0235e60e610ac07506e5/pkg/server/ca/manager.go#L967-L971

Reply:
That isn't the case. `caChain` contains the `CertChain` returned by the upstream CA. When you join upstream PKI, `upstreamChain` is set to `caChain` (preserving the whole chain returned by the upstream CA). `caChain[0]` is the server CA certificate signed by upstream.

Reply:
Ah I see now. So is the full upstreamChain appended to the leaf SVID then?

Reply:
Yes https://github.com/spiffe/spire/blob/master/pkg/server/ca/ca.go#L291